hibernate-jcache/src/main/java/org/hibernate/cache/jcache/internal/JCacheRegionFactory.java

```diff
@@ -43,6 +44,12 @@ public class JCacheRegionFactory extends RegionFactoryTemplate {
 	private volatile CacheManager cacheManager;
 	private volatile MissingCacheStrategy missingCacheStrategy;
 
+	// need to save the classloader from Caching.getDefaultClassLoader to reset it
+	// when this service is released again.
+	private volatile ClassLoader oldJsr107CacheClassLoader;
```

Comment:
Is it really necessary to keep around the old class loader? Can't we just swap the class loader when retrieving/looking-up the `CacheManager`?